### PR TITLE
CI: Use configurable branch for sonobuoy job

### PR DESCRIPTION
**Pull request by Itxaka**
Looks like we were hardcoding the branch value on the job template
for sonobuoy, but what we really want is to have the default
branch parameter be master and allow setting the branch to pick
the jenkinsfile from set to that value dynamically so we
can test changes in the jenkinsfile from PRs.

## Why is this PR needed?

Does it fix an issue? addresses a business case?

add a description and link to the issue if one exists.

Fixes #

**Reminder**: Add the "fixes bsc#XXXX" to the title of the commit so that it will
appear in the changelog.

## What does this PR do?

please include a brief "management" technical overview (details are in the code)

## Anything else a reviewer needs to know?

Special test cases, manual steps, links to resources or anything else that could be helpful to the reviewer.

## Info for QA

This is info for QA so that they can validate this. This is **mandatory** if this PR fixes a bug.
If this is a new feature, a good description in "What does this PR do" may be enough.

### Related info

Info that can be relevant for QA:
* link to other PRs that should be merged together
* link to packages that should be released together
* upstream issues

### Status **BEFORE** applying the patch

How can we reproduce the issue? How can we see this issue? Please provide the steps and the prove
this issue is not fixed.

### Status **AFTER** applying the patch

How can we validate this issue is fixed? Please provide the steps and the prove this issue is fixed.

## Docs

If docs need to be updated, please add a link to a PR to https://github.com/SUSE/doc-caasp.
At the time of creating the issue, this PR can be work in progress (set its title to [WIP]),
but the documentation needs to be finalized before the PR can be merged. 

# Merge restrictions

(Please do not edit this)

We are in *v4-maintenance phase*, so we will restrict what can be merged to prevent unexpected surprises:

    What can be merged (merge criteria):
        2 approvals:
            1 developer: code is fine
            1 QA: QA is fine
        there is a PR for updating documentation (or a statement that this is not needed)

### CODE REVIEW

File: ci/jenkins/templates/conformance-template.yaml [hunk 1]
```diff
@@ -25,8 +24,13 @@
                 url: 'https://github.com/{repo-owner}/{repo-name}.git'
                 credentials-id: '{repo-credentials}'
                 branches:
-                    - '{branch}'
+                    - $branch
```
Comment (pablochacin) [inline ci/jenkins/templates/conformance-template.yaml]: Shouldn't this be `${BRANCH}'?

File: ci/jenkins/templates/conformance-template.yaml [hunk 2]
```diff
@@ -13,7 +12,7 @@
     parameters:
         - string:
             name: BRANCH
-            default: '{branch}'
+            default: master
```
Comment (pablochacin) [inline ci/jenkins/templates/conformance-template.yaml]: not sure if this must have quotes: `"master`"

File: ci/jenkins/templates/conformance-template.yaml (hunk 1 above)
Comment (Itxaka) [reply ci/jenkins/templates/conformance-template.yaml]: if we use `${BRANCH}` then jjb tries to interpolate it and fails due to not finding the branch value anywhere. Using simply `$branch` make jjb ignore it and jenkins use the proper `BRANCH` parameter that we define above :D

File: ci/jenkins/templates/conformance-template.yaml (hunk 2 above)
Comment (Itxaka) [reply ci/jenkins/templates/conformance-template.yaml]: mmmh, good point, lets see what jjb prints

Comment (Itxaka) [reply ci/jenkins/templates/conformance-template.yaml]: no need! jjb output looks good.

File: ci/jenkins/templates/conformance-template.yaml (hunk 1 above)
Comment (aojea) [reply ci/jenkins/templates/conformance-template.yaml]: IIRC for jjb you have to use double brackets on job templates if you want to embed bash variables 
https://docs.openstack.org/infra/jenkins-job-builder/definition.html#macro-notes
Also, I think that is recommended to default the variable always
`${{BRANCH:-}}`

Comment (Itxaka) [reply ci/jenkins/templates/conformance-template.yaml]: nice, thanks for linking to the proper docs, I had a hard time finding that.

**Comment by jordimassaguerpla** [comment]
Already addressed by https://github.com/SUSE/skuba/pull/1094

**Pull request closed without merging**